libopenage/datastructure/doubly_linked_list.h

```diff
@@ -85,6 +85,10 @@ class DoublyLinkedList {
 
 		node->next->previous = new_node;
 		node->next           = new_node;
+
+		if (this->last == node) {
+				this->last = new_node;
+		}
```

Comment:
like insert_after(), I'm not sure, if we want to make insert_*() check for this, or let the caller take responsibility for that case.